VirtoCommerce.CatalogModule.Web/Controllers/Api/CatalogModuleCategoriesController.cs

```diff
@@ -170,5 +175,39 @@ public IHttpActionResult Delete([FromUri]string[] ids)
             _categoryService.Delete(ids);
             return StatusCode(HttpStatusCode.NoContent);
         }
+
+        /// <summary>
+        /// Bulk deletes the specified categories by id.
+        /// </summary>
+        /// <param name="searchCriteria"></param>
+        [HttpPost]
+        [Route("delete")]
+        [ResponseType(typeof(void))]
+        public async Task<IHttpActionResult> BulkDelete(CategorySearchCriteria searchCriteria)
+        {
+            bool hasSearchCriteria;
+            var listIds = new List<string>();
+            do
+            {
+                var searchResult = await _categorySearchService.SearchAsync(searchCriteria);
+
+                hasSearchCriteria = searchResult.Items != null;
+                if (hasSearchCriteria)
+                {
+                    CheckCurrentUserHasPermissionForObjects(CatalogPredefinedPermissions.Delete, searchResult.Items);
+                    searchResult.Items.ForEach(res => listIds.Add(res.Id));
+                    searchCriteria.Skip += searchCriteria.Take;
+
+                }
+            }
+            while (hasSearchCriteria);
+
+            listIds.ProcessWithPaging(searchCriteria.Take, (ids, currentItem, totalCount) =>
```

Comment:
Here I think we should not use `searchCriteria.Take` as batch size. It could be huge number. We should use reasonable value, e.g. _deleteBatchSize = 50.